### PR TITLE
PPS Master Unit Testing Groundwork [CORE-360]

### PR DESCRIPTION
**Pull request by acohen4**

### CODE REVIEW

File: src/server/pps/server/server.go
Comment (acohen4) [inline src/server/pps/server/server.go:48]: will revert this file

File: src/server/pps/server/pipeline_state_driver.go
Comment (PFedak) [inline src/server/pps/server/pipeline_state_driver.go:185]: can't decide if it's important to capture, but strictly speaking `SetPipelineState` will only modify the version of the pipeline at that spec commit, which may no longer be current. This is important for concurrent updates, so I expect something like 

- send pipeline event for a failed pipeline while blocking `getRC` from completing
- update pipeline externally
- unblock `getRC` to let the old step continue with a stale pipeline info

to behave incorrectly in this model, with the pipeline staying in failure instead of going into starting etc

even if we don't want to capture that, this model feels a little weird - you could just be using the repo name from the spec commit, right? Or is the idea to test that the spec commit ID made it into the map properly?

File: src/server/pps/server/pipeline_state_driver.go
Comment (PFedak) [inline src/server/pps/server/pipeline_state_driver.go:160]: what situations are you thinking of where the request/pipeline controller context wouldn't be appropriate? We're already a few function calls deep at this point, I guess

File: src/server/pps/server/pipeline_controller_test.go
Comment (PFedak) [inline src/server/pps/server/pipeline_controller_test.go:232]: ok, I'm definitely missing something - isn't the `assertWhen` going to succeed when the last state is RUNNING? Is it just going back quickly enough that STANDBY shows up anyway?

File: src/server/pps/server/pipeline_controller_test.go
Comment (PFedak) [inline src/server/pps/server/pipeline_controller_test.go:285]: I'd generally prefer to do things like this using channels (or directly checking the condition you're waiting for) if it's not too messy

File: src/server/pps/server/pipeline_controller.go
Comment (PFedak) [inline src/server/pps/server/pipeline_controller.go:654]: update comment

File: src/server/pps/server/pipeline_controller.go
Comment (PFedak) [inline src/server/pps/server/pipeline_controller.go:314]: hmm, feels like we narrowly avoided a bug before in the case of an external RC deletion while RUNNING? `getRC` would return a nil `rc` but `restart` would be `false`, `evaluate` would say we need to restart, `apply` would error on the restart because of how that function worked. It was okay because we errored out of the backoff, too, and would just print the failure at the top level. Still makes me a little nervous

**Review by PFedak — COMMENTED** [review]
I still want to look through the tests you've added to see if there's anything that I'd like to see. I think we can probably delete the `TestStandbyTransitions` test and some of the other `Autoscaling` ones now, which is nice

File: src/server/pps/server/infra_driver.go
Comment (acohen4) [reply src/server/pps/server/infra_driver.go:45]: totally agree 👍

File: src/server/pps/server/pipeline_controller.go
Comment (acohen4) [reply src/server/pps/server/pipeline_controller.go:364]: yeah it does seem silly

File: src/server/pps/server/pipeline_state_driver.go
Comment (acohen4) [reply src/server/pps/server/pipeline_state_driver.go:160]: I actually think this TODO is outdated. The function should be passed the PC's ctx

File: src/server/pps/server/pipeline_controller_test.go
Comment (acohen4) [reply src/server/pps/server/pipeline_controller_test.go:94]: yes that's way better :)

File: src/server/pps/server/pipeline_controller_test.go
Comment (acohen4) [reply src/server/pps/server/pipeline_controller_test.go:72]: No you're right. I'm going to separate out the two cases

File: src/server/pps/server/pipeline_controller_test.go
Comment (acohen4) [reply src/server/pps/server/pipeline_controller_test.go:221]: yeah you're right. I removed `assertWhen` due to discussion above

File: src/server/pps/server/pipeline_state_driver.go
Comment (acohen4) [reply src/server/pps/server/pipeline_state_driver.go]: I changed the name of it here. I was using it to force closing the channel that is otherwise closed by ctx.Done()

File: src/server/pps/server/pipeline_state_driver.go
Comment (acohen4) [reply src/server/pps/server/pipeline_state_driver.go]: yeah the model here should be inverted so that `pipelines` -> `specCommit` -> `pipelineInfo`

File: src/server/pps/server/monitor.go
Comment (acohen4) [reply src/server/pps/server/monitor.go]: I changed this the RetryWithCancel

File: src/server/pps/server/pipeline_controller.go
Comment (acohen4) [reply src/server/pps/server/pipeline_controller.go:314]: right (yikes)

File: src/server/pps/server/pipeline_state_driver.go
Comment (PFedak) [inline src/server/pps/server/pipeline_state_driver.go:232]: could close in a defer to handle both, but not super important

File: src/server/pps/server/pipeline_state_driver.go
Comment (PFedak) [inline src/server/pps/server/pipeline_state_driver.go:289]: can also be a close

**Review by PFedak — APPROVED** [review]
LGTM, just depends if you want to remove the (very slow) integration test versions of the autoscaling/standby tests you're reproducing here

**Pull request merged**
